### PR TITLE
New variable for additional assume role policy principals aws identifiers

### LINKED ISSUE #391: Additional worker IAM role assume role policy
# I have issues

I'm using kiam and it needs trust relationship to kiam server IAM role. But there is no way to specify additional assume role policy for `aws_iam_role.workers`

## I'm submitting a...

* [ ] bug report
* [x] feature request
* [ ] support request
* [ ] kudos, thank you, warm fuzzy

## What is the current behavior?

`awm_iam_role.workers` use `data.aws_iam_policy_document.workers_assume_role_policy.json` and `data.aws_iam_policy_document.workers_assume_role_policy` has `ec2.amazonaws.com` principals only.

## What's the expected behavior?

There should be additional principal variables.

### PR DESCRIPTION
**Pull request by toughrogrammer**
# PR o'clock

## Description

fix #391 

I know it's not very flexible solution but it would be better to use Dynamic Nested Blocks feature when it supports terraform 0.12

### Checklist

- [x] `terraform fmt` and `terraform validate` both work from the root and `examples/*` directories
- [x] CI tests are passing
- [x] I've added my change to CHANGELOG.md and highlighted any breaking changes
- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

File: examples/basic/main.tf [hunk 1]
```diff
@@ -132,4 +132,8 @@ module "eks" {
   map_users_count                      = "${var.map_users_count}"
   map_accounts                         = "${var.map_accounts}"
   map_accounts_count                   = "${var.map_accounts_count}"
+
+  workers_assume_role_policy_principals_aws_identifiers = [
```
Comment (max-rocket-internet) [inline examples/basic/main.tf]: While it's good to have examples to show things are done, the `examples/basic` directory is supposed to be very simple, and I don't think this item falls into that category 🙂

File: data.tf [hunk 2]
```diff
@@ -12,6 +12,11 @@ data "aws_iam_policy_document" "workers_assume_role_policy" {
       type        = "Service"
       identifiers = ["ec2.amazonaws.com"]
     }
+
+    principals {
+      type        = "AWS"
+      identifiers = "${var.workers_assume_role_policy_principals_aws_identifiers}"
```
Comment (max-rocket-internet) [inline data.tf]: According to the variable default, this is a list. But here your value isn't inside `[]`. Is this correct?

**Comment by max-rocket-internet** [comment]
Thanks for the PR @toughrogrammer! Can you also observe the tick boxes? They are there for a reason 😃Thanks

File: data.tf (hunk 2 above)
Comment (toughrogrammer) [reply data.tf]: It works and I found `["${var.workers_assume_role_policy_principals_aws_identifiers}"]` also works. How do you think which one is better for readability? I prefer to not write redundant codes, so I wrote like it.

File: examples/basic/main.tf (hunk 1 above)
Comment (toughrogrammer) [reply examples/basic/main.tf]: I'm concerning... There isn't proper example to handle issue about IAM. But i think it's not good to create new example for only this variable.

**Comment by toughrogrammer** [comment]
> Thanks for the PR @toughrogrammer! Can you also observe the tick boxes? They are there for a reason 😃Thanks

Done! I'm sorry I didn't know that.

File: examples/basic/main.tf (hunk 1 above)
Comment (max-rocket-internet) [reply examples/basic/main.tf]: > i think it's not good to create new example for only this variable.

Not here. Please remove. Like I said, this is supposed to be the most basic example to get a cluster running. If you are including some random and invalid ARN then it's just confusing.

File: data.tf (hunk 2 above)
Comment (max-rocket-internet) [reply data.tf]: Yes, inside `[]` please. It's the standard way in Terraform 0.11. Whether it's readable or makes sense is another question 😅

File: variables.tf [hunk 3]
```diff
@@ -274,3 +274,8 @@ variable "manage_worker_iam_resources" {
   description = "Whether to let the module manage worker IAM resources. If set to false, iam_instance_profile_name must be specified for workers."
   default     = true
 }
+
+variable workers_assume_role_policy_principals_aws_identifiers {
```
Comment (max-rocket-internet) [inline variables.tf]: You also need to be sure TF runs without error when this variable isn't set, i.e the default value.

**Comment by max-rocket-internet** [comment]
I'm gonna close this as the module has now moved to TF 0.12 Syntax and it's been a while. Feel free to reopen 🙂

**Pull request closed without merging**